ssc/cmod_irradproc.cpp

```diff
@@ -29,230 +29,264 @@ OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-			
-			p_azm[i] = (ssc_number_t) solazi;
-			p_zen[i] = (ssc_number_t) solzen;
-			p_elv[i] = (ssc_number_t) solelv;
-			p_dec[i] = (ssc_number_t) soldec;	
-			p_sunrise[i] = (ssc_number_t) sunrise;
-			p_sunset[i] = (ssc_number_t) sunset;
-			p_sunup[i] = (ssc_number_t) sunup;
-			
-			
-			double aoi, stilt, sazi, rot, btd;
-			x.get_angles( &aoi, &stilt, &sazi, &rot, &btd );
-
-			// assign outputs
-			p_inc[i] = (ssc_number_t) aoi;
-			p_surftilt[i] = (ssc_number_t) stilt;
-			p_surfazm[i] = (ssc_number_t) sazi;
-			p_rot[i] = (ssc_number_t) rot;
-			p_btdiff[i] = (ssc_number_t) btd;
-			
-			double beam, skydiff, gnddiff, iso, cir, hor;
-			x.get_poa( &beam, &skydiff, &gnddiff, &iso, &cir, &hor );
-
-			p_poa_beam[i] = (ssc_number_t) beam;
-			p_poa_skydiff[i] = (ssc_number_t) skydiff;
-			p_poa_gnddiff[i] = (ssc_number_t) gnddiff;
-			p_poa_skydiff_iso[i] = (ssc_number_t) iso;
-			p_poa_skydiff_cir[i] = (ssc_number_t) cir;
-			p_poa_skydiff_hor[i] = (ssc_number_t) hor;
-			
-		}
-	}
+    cm_irradproc()
+    {
+        add_var_info(_cm_vtab_irradproc);
+    }
+
+    void exec()
+    {
+        size_t count;
+        ssc_number_t* beam = 0, * glob = 0, * diff = 0;
+        int irrad_mode = as_integer("irrad_mode");
+        if (irrad_mode == 0) //beam and diffuse
+        {
+            beam = as_array("beam", &count);
+            if (count < 2) throw general_error("need at least 2 data points in irradproc");
+            diff = as_array("diffuse", &count);
+        }
+        else if (irrad_mode == 1) //global and beam
+        {
+            beam = as_array("beam", &count);
+            if (count < 2) throw general_error("need at least 2 data points in irradproc");
+            glob = as_array("global", &count);
+        }
+        else //global and diffuse
+        {
+            diff = as_array("diffuse", &count);
+            if (count < 2) throw general_error("need at least 2 data points in irradproc");
+            glob = as_array("global", &count);
+        }
+
+        ssc_number_t* year = as_array("year", &count);
+        ssc_number_t* month = as_array("month", &count);
+        ssc_number_t* day = as_array("day", &count);
+        ssc_number_t* hour = as_array("hour", &count);
+        ssc_number_t* minute = as_array("minute", &count);
+
+        int sky_model = as_integer("sky_model");
+
+        double lat = as_double("lat");
+        double lon = as_double("lon");
+        double tz = as_double("tz");
+        double elev, tdry, pres;
+
+        if (!is_assigned("elevation")) {
+            elev = 0; //assume 0 meter elevation if none is provided
+        }
+        else {
+            elev = as_double("elevation");
+            if (elev < 0 || elev > 5100) {
+                elev = elev / 3.281; //adjust feet to meters if greater than the highest city (La Rinoconada, Peru)
+                throw exec_error("poacalib", "The elevation input is outside of the expected range. Please make sure that the units are in meters");
```

Comment:
the first parameter to `exec_error` is the name of the compute_module, in this case `irradproc`